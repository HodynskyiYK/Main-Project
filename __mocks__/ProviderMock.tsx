import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}