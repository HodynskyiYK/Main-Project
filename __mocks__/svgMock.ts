// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock