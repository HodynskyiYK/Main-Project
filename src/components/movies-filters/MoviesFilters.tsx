import React, {FC} from 'react'
import {IFilter, IMoviesFilters} from './filtersTypes'

export const MoviesFilters: FC<IMoviesFilters> = ({filters}) => {

    return (
        <ul className="filters-list">{
            filters.map((filter: IFilter) => (
                <li key={filter.id} className="filters-item">
                    <span
                        className={filter.active ? 'filter-value active' : 'filter-value'}
                    >{filter.name}</span>
                </li>
            ))
        }</ul>
    )
}
