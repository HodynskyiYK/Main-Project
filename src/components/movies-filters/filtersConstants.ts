export const FILTERS_VALUES = [
    {
        name: 'All',
        value: 'all',
        active: true,
        id: 1
    },
    {
        name: 'Documentary',
        value: 'documentary',
        active: false,
        id: 2
    },
    {
        name: 'Comedy',
        value: 'comedy',
        active: false,
        id: 3
    },
    {
        name: 'Horror',
        value: 'horror',
        active: false,
        id: 4
    },
    {
        name: 'Crime',
        value: 'crime',
        active: false,
        id: 5
    }
]