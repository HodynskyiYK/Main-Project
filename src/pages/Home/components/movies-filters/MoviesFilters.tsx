import React, { FC, useState } from 'react'
import classnames from 'classnames'
import {IFilter, IMoviesFilters} from './filtersTypes'
import styles from './MoviesFilters.module.scss'
import { useAction, useTypedSelector } from '../../../../hooks'

export const MoviesFilters: FC<IMoviesFilters> = ({filters}) => {
    const [activeFilter, setActiveFilter] = useState<string>('All')
    const {fetchMovies} = useAction()
    const {sortBy} = useTypedSelector(state => state.movies)

    const filterMovies = (genre: string) => {
        if (genre !== activeFilter) {
            const genreForFilters = genre !== 'All' ? genre : ''
            fetchMovies(genreForFilters, sortBy)
            setActiveFilter(genre)
        }
    }

    return (
        <ul className={styles.filtersList}>{
            filters.map((filter: IFilter) => (
                <li key={filter.id} className={styles.filtersItem}>
                    <span
                        className={filter.value === activeFilter ?
                            classnames(styles.filterValue, styles.activeFilter) :
                            styles.filterValue}
                        onClick={() => filterMovies(filter.value)}
                    >{filter.name}</span>
                </li>
            ))
        }</ul>
    )
}
