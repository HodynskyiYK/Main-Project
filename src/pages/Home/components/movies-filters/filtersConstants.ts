export const FILTERS_VALUES = [
    {
        name: 'All',
        value: 'All',
        active: true,
        id: 1
    },
    {
        name: 'Documentary',
        value: 'Documentary',
        active: false,
        id: 2
    },
    {
        name: 'Comedy',
        value: 'Comedy',
        active: false,
        id: 3
    },
    {
        name: 'Horror',
        value: 'Horror',
        active: false,
        id: 4
    },
    {
        name: 'Crime',
        value: 'Crime',
        active: false,
        id: 5
    }
]