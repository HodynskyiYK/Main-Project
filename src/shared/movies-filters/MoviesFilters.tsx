import React, { FC, useEffect, useState } from 'react'
import classnames from 'classnames'
import {IFilter, IMoviesFilters} from './filtersTypes'
import styles from './MoviesFilters.module.scss'
import { useAction, useTypedSelector } from '../../hooks'

export const MoviesFilters: FC<IMoviesFilters> = ({filters}) => {
    const [activeFilter, setActiveFilter] = useState<string>('All')
    const {fetchMovies} = useAction()
    const {sortBy, filterBy} = useTypedSelector(state => state.movies)

    useEffect(() => {
        if (filterBy) {
            setActiveFilter(filterBy)
        }
    }, [filterBy])

    const filterMovies = (genre: string) => {
        if (genre !== activeFilter) {
            const genreForFilters = genre !== 'All' ? genre : ''
            fetchMovies(genreForFilters, sortBy)
            setActiveFilter(genre)
        }
    }

    return (
        <ul className={styles.filtersList}>{
            filters.map((filter: IFilter) => (
                <li key={filter.id} className={styles.filtersItem}>
                    <span
                        className={filter.value === activeFilter ?
                            classnames(styles.filterValue, styles.activeFilter) :
                            styles.filterValue}
                        onClick={() => filterMovies(filter.value)}
                    >{filter.name}</span>
                </li>
            ))
        }</ul>
    )
}
