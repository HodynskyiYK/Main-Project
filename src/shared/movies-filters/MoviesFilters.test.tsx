import React from 'react'
import { render } from '@testing-library/react'
import { MoviesFilters } from './MoviesFilters'
import { FILTERS_VALUES } from './filtersConstants'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('MoviesFilters tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock>
            <MoviesFilters filters={FILTERS_VALUES}/>
        </ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})