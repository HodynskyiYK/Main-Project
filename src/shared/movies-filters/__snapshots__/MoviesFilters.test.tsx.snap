// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesFilters tests To match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="filtersList"
  >
    <li
      class="filtersItem"
    >
      <span
        class="filterValue activeFilter"
      >
        All
      </span>
    </li>
    <li
      class="filtersItem"
    >
      <span
        class="filterValue"
      >
        Documentary
      </span>
    </li>
    <li
      class="filtersItem"
    >
      <span
        class="filterValue"
      >
        Comedy
      </span>
    </li>
    <li
      class="filtersItem"
    >
      <span
        class="filterValue"
      >
        Horror
      </span>
    </li>
    <li
      class="filtersItem"
    >
      <span
        class="filterValue"
      >
        Crime
      </span>
    </li>
    <li
      class="filtersItem"
    >
      <span
        class="filterValue"
      >
        Action
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
